client/packages_test.go

```diff
@@ -185,6 +185,18 @@ func (cs *clientSuite) TestClientFindPrefix(c *check.C) {
 	c.Check(cs.req.URL.RawQuery, check.Equals, "name=foo%2A") // 2A is `*`
 }
 
+func (cs *clientSuite) TestClientFindCommonID(c *check.C) {
+	_, _, _ = cs.cli.Find(&client.FindOptions{CommonID: "org.kde.ktuberling.desktop"})
+	c.Check(cs.req.URL.Path, check.Equals, "/v2/find")
+	c.Check(cs.req.URL.RawQuery, check.Equals, "common-id=org.kde.ktuberling.desktop")
+}
+
+func (cs *clientSuite) TestClientFindCommonIDAndQuery(c *check.C) {
+	_, _, _ = cs.cli.Find(&client.FindOptions{CommonID: "org.kde.ktuberling.desktop", Query: "potato"})
+	c.Check(cs.req.URL.Path, check.Equals, "/v2/find")
+	c.Check(cs.req.URL.RawQuery, check.Equals, "common-id=org.kde.ktuberling.desktop&q=potato")
```

Comment:
When will people use common-id=... and q=... together? Seems to me like common-id is a very specific search, how would that be combined with other queries?

Reply:
if there were 137 `io.snapcraft.hello` apps, and you wanted the one that said `mvo` in the summary?

It's a stretch, I know. But all the parts support it, so why not :man_shrugging:

Reply:
It think we'll not support this in snapd-glib, to avoid making the API explode, i.e.
```
snapd_client_find_sync (client, SNAPD_FIND_FLAGS_NONE, "fo", NULL, cancellable, &error);
snapd_client_find_sync (client, SNAPD_FIND_FLAGS_MATCH_NAME, "foo", NULL, cancellable, &error);
snapd_client_find_sync (client, SNAPD_FIND_FLAGS_MATCH_COMMON_ID, "com.example.foo", NULL, cancellable, &error);
```
These do:
```
/v2/find?q=fo
/v2/find?name=foo
/v2/find?common-id=com.example.foo
```

Reply:
as to my "why not": because it's easier to add support than to remove it. As it's not going to be used, I dropped support for the combo.